contracts/interchain-token-service/src/contract.rs

```diff
@@ -260,15 +260,15 @@ impl AxelarExecutableInterface for InterchainTokenService {
     ) {
         let _ = Self::validate_message(&env, &source_chain, &message_id, &source_address, &payload);
 
-        let _ = Self::execute_message(
+        event::executed(
             &env,
             source_chain.clone(),
             message_id.clone(),
             source_address.clone(),
             payload.clone(),
         );
 
-        event::executed(&env, source_chain, message_id, source_address, payload);
+        let _ = Self::execute_message(&env, source_chain, message_id, source_address, payload);
```

Comment:
`let _ ` is a bug. It's ignoring the error being returned by `execute_message` which will prevent tx from reverting. Instead, return the error. Since this is an interface/trait method, the error type needs to be parameterized on the trait. See `MigratableInterface` for reference

Reply:
Handled in separate PR